src/parquet/arrow/reader.h

```diff
@@ -149,6 +154,21 @@ class PARQUET_EXPORT FileReader {
   ::arrow::Status ReadSchemaField(int i, const std::vector<int>& indices,
                                   std::shared_ptr<::arrow::Array>* out);
 
+  /// \brief Return a RecordBatchReader of row groups selected from row_group_indices, the
+  ///    ordering in row_group_indices matters.
+  /// \returns error Status if row_group_indices contains invalid index
+  ::arrow::Status GetRecordBatchReader(const std::vector<int>& row_group_indices,
+                                       std::shared_ptr<::arrow::RecordBatchReader>* out);
+
+  /// \brief Return a RecordBatchReader of row groups selected from row_group_indices,
+  ///     whose columns are selected by column_indices. The ordering in row_group_indices
+  ///     and column_indices matter.
+  /// \returns error Status if either row_group_indices or column_indices contains invalid
+  ///    index
+  ::arrow::Status GetRecordBatchReader(const std::vector<int>& row_group_indices,
+                                       const std::vector<int>& column_indices,
+                                       std::shared_ptr<::arrow::RecordBatchReader>* out);
```

Comment:
My main critique of these APIs is that we will want to provide for setting the number of rows to be read for each call to `ReadNext`, for example 1,000,000 rows at a time. Right now this is returning a whole row group at a time

Reply:
Could you please open a JIRA about improving this code in this regard?

Reply:
>  My main critique of these APIs is that we will want to provide for setting the number of rows to be read for each call to `ReadNext`

Ah, I do consider this when implementing this. However the [`RecordBatch`](https://github.com/apache/arrow/blob/master/cpp/src/arrow/record_batch.h#L166) interface in arrow doesn't expose that. And I'd like to hide impl details in `parquet/arrow/reader`. To enable this, I'd like to proposal new method to `RecordBatch` then.
What do you think? @wesm 

> Could you please open a JIRA about improving this code in this regard?

Will do.